controllers/etcdpeer_controller.go

```diff
@@ -587,13 +596,132 @@ func (r *EtcdPeerReconciler) updateHealth(peer *etcdv1alpha1.EtcdPeer, healthy b
 	peer.Status.Healthy = healthy
 }
 
+func (r *EtcdPeerReconciler) updateLastHealthy(peer *etcdv1alpha1.EtcdPeer, lastHealthy time.Time) {
+	peer.Status.LastHealthy = metav1.NewTime(lastHealthy)
+}
+
+func (r *EtcdPeerReconciler) getPVCForPeer(ctx context.Context, peer *etcdv1alpha1.EtcdPeer) (*corev1.PersistentVolumeClaim, error) {
+	expectedPvc := pvcForPeer(peer)
+	expectedPvcNamespacedName := client.ObjectKeyFromObject(expectedPvc)
+
+	var actualPvc corev1.PersistentVolumeClaim
+	err := r.Client.Get(ctx, expectedPvcNamespacedName, &actualPvc)
+	if err != nil {
+		return nil, err
+	}
+
+	return &actualPvc, nil
+}
+
+func (r *EtcdPeerReconciler) findPodForPeer(ctx context.Context, peer *etcdv1alpha1.EtcdPeer) (*corev1.Pod, error) {
+	var podList corev1.PodList
+	err := r.Client.List(ctx, &podList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			peerLabel:    peer.Name,
+			clusterLabel: peer.Spec.ClusterName,
+		}),
+		Namespace: peer.Namespace,
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list pods")
+	}
+
+	if len(podList.Items) < 1 {
+		return nil, errors.New("cannot find any pods for peer")
+	} else if len(podList.Items) > 1 {
+		return nil, fmt.Errorf("found more than one pod for peer: %v", podList.Items)
+	}
+	return &podList.Items[0], nil
+}
+
+func (r *EtcdPeerReconciler) findVolumeAttachmentForPeer(ctx context.Context, peer *etcdv1alpha1.EtcdPeer) (*storage.VolumeAttachment, error) {
+	pvc, err := r.getPVCForPeer(ctx, peer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find pvc whilst looking for volume attachment")
+	}
+
+	volumeAttachments := &storage.VolumeAttachmentList{}
+	err = r.Client.List(ctx, volumeAttachments, &client.ListOptions{
+		// VolumeAttachments aren't namespaced, so this needs to be here and it needs to be an empty string
+		Namespace: "",
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list volume attachments")
+	}
+
+	for _, va := range volumeAttachments.Items {
+		if va.Spec.Source.PersistentVolumeName != nil && *va.Spec.Source.PersistentVolumeName == pvc.Spec.VolumeName {
+			return &va, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to find volume attachment for pvc: %v", pvc)
```

Comment:
Couldn't this be a silent fail as there's nothing to do (k8s perhaps was faster) and thus we just fall through from here?

Reply:
I would argue that this is an error case for the function `findVolumeAttachmentForPeer`. 

It might not matter for how we use the function at the moment, but it could matter in the future.